interpreter/terminal_interface/magic_commands.py

```diff
@@ -72,47 +82,131 @@ def handle_debug(self, arguments=None):
+                    )
+                    display_markdown_message(f"**'{no_container_message}'**")
+            else:
+                file_not_found_message = f"File `{filepath}` does not exist."
+                display_markdown_message(file_not_found_message)
+    else:
+        ignore_command_message = "File uploads are only used when using containerized code execution. Ignoring command."
+        display_markdown_message(f"**{ignore_command_message}**")
+        
+def handle_container_download(self, *args):
+    if self.use_containers:
+        try:
+            client = docker.APIClient()
+        except Exception as e:
+            print("[BOLD][RED]Unable to connect to the Docker Container daemon. Please ensure Docker is installed and running. ignoring command[/BOLD]")
+            return
+        
+        session_id = SESSION_IDS_BY_OBJECT.get(self)
+        if session_id is None:
+            print("No session found. Please run any code to start one.")
+            return
+
+        containers = client.containers(filters={"label": f"session_id={session_id}"})
+        if not containers:
+            print("No container found to download from. Please run any code to start one.")
+            return
+        
+        container_id = containers[0]['Id']
+
+        # Define the local directory where the files will be downloaded.
+        # Using 'Open Interpreter' as the appname and no author.
+        local_dir = appdirs.user_data_dir(appname="Open Interpreter")
+
+        for file_path_in_container in args:
+            # Construct the local file path
+            local_file_path = os.path.join(local_dir, os.path.basename(file_path_in_container))
+            
+            # Attempt to download the file and handle exceptions
+            try:
+                download_file_from_container(container_id, file_path_in_container, local_file_path)
+                print(f"File downloaded to {local_file_path}")
+            except docker.errors.NotFound:
+                print(f"File {file_path_in_container} not found in the container.")
+    else:
+        print("File downloads are only used when using containerized code execution. Ignoring command.")
 
-    display_markdown_message(f"> messages json loaded from {os.path.abspath(json_path)}")
 
 def handle_magic_command(self, user_input):
     # split the command into the command and the arguments, by the first whitespace
     switch = {
-      "help": handle_help,
-      "debug": handle_debug,
-      "reset": handle_reset,
-      "save_message": handle_save_message,
-      "load_message": handle_load_message,
-      "undo": handle_undo,
+        "help": handle_help,
+        "debug": handle_debug,
+        "reset": handle_reset,
+        "save_message": handle_save_message,
+        "load_message": handle_load_message,
+        "undo": handle_undo,
+        "upload": handle_container_upload,
+        "download": handle_container_download,
```

Comment:
We should also conditionally add these to the output of the `%help` command if running with `use_containers` enabled.